segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/AutoScaleProcessor.java

```diff
@@ -178,19 +179,37 @@ private void triggerScaleDown(String streamSegmentName, boolean silent) {
     }
 
     private CompletableFuture<Void> writeRequest(ScaleEvent event) {
+        if (writer.get() == null) {
+            log.warn("Event writer not set up yet. Ignoring the request");
+            return CompletableFuture.completedFuture(null);
+        }
+
         CompletableFuture<Void> result = new CompletableFuture<>();
+        CompletableFuture<Void> writeComplete = new CompletableFuture<>();
         try {
-            CompletableFuture.runAsync(() -> {
-                try {
-                    writer.get().writeEvent(event.getKey(), event).get();
-                    result.complete(null);
-                } catch (InterruptedException | ExecutionException e) {
-                    log.error("error sending request to requeststream {}", e);
+            AckFuture ackFuture = writer.get().writeEvent(event.getKey(), event);
+            ackFuture.addListener(() -> writeComplete.complete(null), executor);
```

Comment:
This is actually incorrect. And I see why AckFuture is a bad interface... So hopefully we can get a replacement soon but get() still needs to be called inside the runnable as it could have failed.